mozilla-release/mobile/android/chrome/content/browser.js

```diff
@@ -6316,6 +6316,76 @@ var ExternalApps = {
   },
 };
 
+var Cliqz = {
+
+  init: function () {
+    this.oldQuery = "";
+    this.prevPanel = null;
+    // This may file due to delayed addon install that occur on Android
+    this._resolveAddon().then(addon => this._wireAddon(addon));
+  },
+
+  _resolveAddon: function() {
+    var promise = new Promise(resolve => {
+      function getAddon() {
+        AddonManager
+          .getAddonByID("search@cliqz.com")
+          .then(addon => {
+            if (addon) {
+              resolve(addon);
+            } else {
+              console.log("Addon not found");
+              setTimeout(getAddon.bind(this), 1000);
+            }
+          });
+      };
+      getAddon();
+    });
+    return promise;
+  },
+
+  _wireAddon: function(addon) {
+    const json = Services.prefs.getStringPref("extensions.webextensions.uuids", "{}");
```

Comment:
you don't actually need the addon here. You only need the ID which you already know. The whole getAddon is redundant